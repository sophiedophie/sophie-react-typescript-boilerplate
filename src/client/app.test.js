import App from './app';
import React from 'react';
import { shallow } from 'enzyme';

describe('App', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<App />)

    expect(wrapper.find('h1').text()).toBe('Welcome to React Boilerplate App~');
    expect(wrapper.find('p').text()).toBe('Made by sophiedophie👩🏻‍💻');
    expect(wrapper).toMatchSnapshot();
  })
})