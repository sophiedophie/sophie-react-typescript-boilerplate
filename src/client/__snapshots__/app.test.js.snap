// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="app-header-title"
        >
          Welcome to React Boilerplate App~
        </h1>,
        <p
          className="app-header-content"
        >
          Made by 
          <a
            href="https://github.com/sophiedophie"
          >
            sophiedophie👩🏻‍💻
          </a>
        </p>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to React Boilerplate App~",
          "className": "app-header-title",
        },
        "ref": null,
        "rendered": "Welcome to React Boilerplate App~",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Made by ",
            <a
              href="https://github.com/sophiedophie"
            >
              sophiedophie👩🏻‍💻
            </a>,
          ],
          "className": "app-header-content",
        },
        "ref": null,
        "rendered": Array [
          "Made by ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "sophiedophie👩🏻‍💻",
              "href": "https://github.com/sophiedophie",
            },
            "ref": null,
            "rendered": "sophiedophie👩🏻‍💻",
            "type": "a",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="app-header-title"
          >
            Welcome to React Boilerplate App~
          </h1>,
          <p
            className="app-header-content"
          >
            Made by 
            <a
              href="https://github.com/sophiedophie"
            >
              sophiedophie👩🏻‍💻
            </a>
          </p>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to React Boilerplate App~",
            "className": "app-header-title",
          },
          "ref": null,
          "rendered": "Welcome to React Boilerplate App~",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Made by ",
              <a
                href="https://github.com/sophiedophie"
              >
                sophiedophie👩🏻‍💻
              </a>,
            ],
            "className": "app-header-content",
          },
          "ref": null,
          "rendered": Array [
            "Made by ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sophiedophie👩🏻‍💻",
                "href": "https://github.com/sophiedophie",
              },
              "ref": null,
              "rendered": "sophiedophie👩🏻‍💻",
              "type": "a",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
